lib/Partitioner/PartitionerValidation.cpp

```diff
@@ -32,6 +33,9 @@ Error logicalDevicesValidation(
       partitionsNum[backendName].insert(logicalIDList[i]);
     }
     auto backendNum = backendMap.at(backendName).num;
+    if (partitionsNum[backendName].size() > backendNum) {
+      logPartitionInfo(partitions);
+    }
```

Comment:
nit: Since the error has been detected already, next line "RETURN_ERR_IF_NOT" could be an else branch and just return the error.